provider/pkg/provider/helm/v4/values.go

```diff
@@ -93,14 +113,14 @@ func readAsset(p getter.Providers, asset pulumi.Asset) ([]byte, error) {
 	}
 }
 
-// marshalValues converts Pulumi values to Helm values.
-// - Expands assets to their content (to support --set-file).
-func marshalValues(p getter.Providers, a map[string]interface{}) (map[string]interface{}, error) {
+// readAssets converts Pulumi values to Helm values, hydrating Asset values
+// along the way.
+func readAssets(p getter.Providers, a map[string]interface{}) (map[string]interface{}, error) {
```

Comment:
I feel like `marshalValues` is a better name because it isn't just dealing with assets; imagine that resource references were marshaled in some way.